### ISSUE TITLE
[EN][NPC][Sholazar Basin] Primordial Drake

### ISSUE THREAD

**Issue by TheRealestMcCoy** [dalaranwow/dalaran-wow#4363]
**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
Some are the drakes are walking in the air.
**Expected behaviour**: Tell us what should happen instead
They should be flying.
**Steps to reproduce the problem**:

1. Go to Sholazar Basin
2. Go to around 43,27
3. 

**Include proofs for this behaviour**
![wowscrnshot_013017_175332](https://cloud.githubusercontent.com/assets/12252441/22445354/0b08caf6-e715-11e6-86f0-8de6701fea0a.jpg)
![wowscrnshot_013017_175405](https://cloud.githubusercontent.com/assets/12252441/22445375/22c453b8-e715-11e6-894b-cbbf629b47cf.jpg)
![wowscrnshot_013017_175416](https://cloud.githubusercontent.com/assets/12252441/22445377/25caa396-e715-11e6-83a2-6ab35fb54e18.jpg)

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
[Primordial Drake](http://www.wowhead.com/npc=28378/primordial-drake)
You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Rushor** [comment]
some drakes or all of them?

**Comment by Rushor** [comment]
DB/Creature: Sholazar Basin - Primordial Drake - Add hover mode and increase randommovement radius

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4363
* commit: d56fe61713f68fdf18ca41971bf6b57858faa469

![GitHub Logo](http://i.imgur.com/UETHAoH.jpg)

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by Teridan-dal-wow** [comment]
Hello, there are few more things relevent to this report:
- all of the flying drakes have paths, which are missing
- there is a missing mini event in which a drakes attacks an egg and "hatches" it after 2 cast (check end of video)

Video from retail legion from 26.02.2017:
https://www.youtube.com/watch?v=VCj296w8fXA&feature=youtu.be

**Comment by Teridan-dal-wow** [comment]
What is done here:
- added the missing paths for the drakes
- removed the static spawns of npc http://wotlk.openwow.com/npc=28389 (as they should spawn only for the eggs)
- set respawn timer for eggs at 150secs (down from 5min,as it seems the corpse decay on the server for normal creatures seems to be set to 30sec) timer tested on retail (there corplse of egg despawns after 10 secs and new one spawns after 170secs)

Tried to:
- remove the hp regeneration of npc http://wotlk.openwow.com/npc=28408 (currently they engage the player (when attacked), however they don't move or attack, they do keep aggro on player and if you have target of target enabled you will see they target the player. The should:
1) keep the player in combat for 5 secs (the pvp rule),
2) not targer or gain any aggro for the player attacking them (they do have the `extra_flag =2` but still do),
3) not regenerate hp on leaving combat/reset.

To do:
- add the mini event - only the drakes on the ground are part of that event and the range is big (40 yards)
- change the location of where http://wotlk.openwow.com/npc=28389 spawns after an egg is killed. (they use spell http://wotlk.openwow.com/spell=51595 which has 5 yards range however the spawned hatchling should be at 2 yards `Radius: 2 yards` )
- fix the issue with the flying drakes (when you pull them on the group they don't land at all)
![image](https://cloud.githubusercontent.com/assets/6797611/23360775/f4b18da0-fcf5-11e6-8776-b8c68c63866a.png)
- revisit the spawn locations of each drake (the ground ones only) and each egg, as it seems there aren't enough eggs and the possition of some drakes is off by bit.

``` sql
-- delete static hatchlings
DELETE FROM `creature` WHERE `guid` BETWEEN 117803 AND 117808;
DELETE FROM `creature_addon` WHERE `guid` BETWEEN 117803 AND 117808;

UPDATE `creature_template` SET `RegenHealth` = 0 WHERE `entry` = 28408; 
UPDATE `creature` SET `spawntimesecs` = 150 WHERE `guid` BETWEEN 119386 AND 119398;
UPDATE `creature` SET `spawndist` = 10 WHERE `guid` = 100705; 


-- Pathing for Primordial Drake Entry: 28378 'TDB FORMAT' 
SET @NPC := 116787;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6454.238,`position_y`=5019.534,`position_z`=-37.169 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,33554432,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6454.238,5019.534,-37.169,0,0,1,0,100,0), -- 12:14:48
(@PATH,2,6542.277,5002.340,-37.169,0,0,1,0,100,0), -- 12:14:49
(@PATH,3,6514.332,4944.665,-37.169,0,0,1,0,100,0);

-- Pathing for Primordial Drake Entry: 28378 'TDB FORMAT' 
SET @NPC := 116785;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,33554432,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6413.2587,4997.746,-37.169,0,0,1,0,100,0), -- 12:14:48
(@PATH,2,6467.165,5016.772,-37.169,0,0,1,0,100,0), -- 12:14:49
(@PATH,3,6491.813,4956.219,-37.169,0,0,1,0,100,0),
(@PATH,4,6445.914,4943.069,-37.169,0,0,1,0,100,0);

-- Pathing for Primordial Drake Entry: 28378 'TDB FORMAT' 
SET @NPC := 116792;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,33554432,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6510.723,4860.694,-25.966,0,0,1,0,100,0), -- 12:14:48
(@PATH,2,6577.342,4835.685,-25.966,0,0,1,0,100,0), -- 12:14:49
(@PATH,3,6549.686,4765.163,-25.966,0,0,1,0,100,0);

-- Pathing for Primordial Drake Entry: 28378 'TDB FORMAT' 
SET @NPC := 116780;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,33554432,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6547.758,4772.740,-22.966,0,0,1,0,100,0), -- 12:14:48
(@PATH,2,6556.495,4826.752,-22.966,0,0,1,0,100,0), -- 12:14:49
(@PATH,3,6604.082,4757.409,-13.966,0,0,1,0,100,0);

-- Pathing for Primordial Drake Entry: 28378 'TDB FORMAT' 
SET @NPC := 116779;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6538.005,`position_y`=4755.906,`position_z`=-14.124 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,33554432,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6538.005,4755.906,-14.124,0,0,1,0,100,0), -- 12:14:48
(@PATH,2,6560.101,4816.659,-14.124,0,0,1,0,100,0), -- 12:14:49
(@PATH,3,6482.388,4824.500,-14.124,0,0,1,0,100,0),
(@PATH,4,6456.986,4796.889,-14.124,0,0,1,0,100,0);

-- Pathing for Primordial Drake Entry: 28378 'TDB FORMAT' 
SET @NPC := 116781;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,33554432,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6410.162,4768.872,-47.985,0,0,1,0,100,0), -- 12:14:48
(@PATH,2,6437.130,4742.900,-47.985,0,0,1,0,100,0), -- 12:14:49
(@PATH,3,6489.317,4771.561,-47.985,0,0,1,0,100,0),
(@PATH,4,6476.530,4834.381,-47.985,0,0,1,0,100,0);
```

**Comment by Rushor** [comment]
the eggs are already fixed in an other ticket (on PTR already)

**Comment by Teridan-dal-wow** [comment]
Updated the post above, also the cooldown of the mini event for each is 1 min.
I have done what i can for now.

**Comment by Rushor** [comment]
```sql
-- egg flag updates 
UPDATE `creature_template` SET `unit_flags`=131076 WHERE `entry`=28408;
UPDATE `creature_template` SET `flags_extra`=2 WHERE `entry`=28408;
```

and the inhabittype change for the ground npcs with randommovement

```sql
-- update inhabityype for creatures on ground individual
UPDATE `creature` SET `saiscriptflag`=1 WHERE  `guid` IN (116783, 116784, 116786, 116788, 116789, 116790, 116791, 5005152,5005153,5005154,5005156,5005155);
DELETE FROM `smart_scripts` WHERE `entryorguid` IN (-116783, -116784, -116786, -116788, -116789, -116790, -116791, -5005152,-5005153,-5005154,-5005156,-5005155);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(-116783, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-116784, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-116786, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-116788, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-116789, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-116790, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-116791, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-5005152, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-5005153, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-5005154, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-5005156, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
(-5005155, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3');
```

**Comment by Rushor** [comment]
DB/Creature: Primordial Drake and Primordial Drake Egg - Add pathing and individual inhabittype + update behaviour for Primordial Drake Egg 

* updates: https://github.com/dalaranwow/dalaran-wow/issues/4363
* commit: 74e4672079e2c1ec87bd3a36276053d7ad2fbd40

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by Teridan-dal-wow** [comment]
``` sql
(-116783, 0, 0, 0, 25, 0, 100, 0, 0, 0, 0, 0, 206, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Primordial Drake - On Reset - Set Inhabittype 3'),
```
Wouldn't that mean that their inhabittype will be changed only on reset? What about on spawn?

**Comment by Rushor** [comment]
on spawn also triggers the reset action
